pandas/_libs/reduction.pyx

```diff
@@ -134,21 +135,34 @@ cdef class Reducer:
                     res = self.f(chunk)
 
                 # TODO: reason for not squeezing here?
-                res = _extract_result(res, squeeze=False)
+                extracted_res = _extract_result(res, squeeze=False)
                 if i == 0:
                     # On the first pass, we check the output shape to see
                     #  if this looks like a reduction.
-                    _check_result_array(res, len(self.dummy))
-
-                PyArray_SETITEM(result, PyArray_ITER_DATA(it), res)
+                    #  If it does not, return the computed value to be used by the
```

Comment:
this was already patched once, please show a test that is failing now that actually needs this.

Reply:
Can you point to where was this patched?
I know this was fixed for groupby.apply but this is a different issue

please see - 
https://github.com/pandas-dev/pandas/issues/33879

i will add a test